### PR TITLE
Upgrade to grails 4.0.5

### PR DESCRIPTION
**Pull request by charvolant**
Updated to use grails 4.0.5, current versions of ala-auth, ala-bootstrap3 and image-client plugins.
Suitable for use with JDK8
Collects and caches collectory metadata #14
Allows images from data resources as well as collections

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+    // Javascript libraries
+    // compile 'org.webjars:leaflet:1.6.0'
+    compile 'org.webjars:knockout:3.5.1'
+    // compile 'org.webjars:swagger-ui:3.18.2'
+
+    // ALA Plugins
+    compile group: 'org.grails.plugins', name: 'ala-auth', version: '3.2.2'
+    compile group: 'org.grails.plugins', name: 'ala-admin-plugin', version: '2.2'
+    compile group: 'org.grails.plugins', name: 'ala-bootstrap3', version: '3.2.2'
```
Comment (ansell) [inline build.gradle:82]: @charvolant Could you bump this to 3.2.3 to pickup the newly released version that we are deploying for https://github.com/AtlasOfLivingAustralia/ala-infrastructure/issues/756 ?

**Review by ansell — CHANGES_REQUESTED** [review]
Bump ala-bootstrap3 plugin version that was recently released.

File: build.gradle (hunk 1 above)
Comment (charvolant) [reply build.gradle]: Upgraded

File: grails-app/conf/application.yml [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+    pooled: true
+    jmxExport: true
+    driverClassName: org.h2.Driver
+    username: sa
+    password: ''
+
+environments:
+    development:
+        dataSource:
+            dbCreate: create-drop
```
Comment (Rita-C) [inline grails-app/conf/application.yml:115]: could `update` be enough?

File: grails-app/init/specimenbrowser/BootStrap.groovy [hunk 3]
```diff
@@ -1,3 +1,5 @@
+package specimenbrowser
```
Comment (Rita-C) [inline grails-app/init/specimenbrowser/BootStrap.groovy:1]: not `package au.org.ala.specimenbrowser`?

File: grails-app/services/au/org/ala/specimenbrowser/ResourceService.groovy [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    /**
+     * Get the path to the resource in the collectory, based on the resource UID prefix
+     *
+     * @param uid The resource UID (eg. co12, dr1456)
+     * @return The path to that resource (eg collection, dataResource)
+     */
+    String resourcePath(String uid) {
+        def mapping = resourceMappings.find { m -> uid.startsWith(m.prefix) }
+        if (mapping)
+            return mapping.path
```
Comment (Rita-C) [inline grails-app/services/au/org/ala/specimenbrowser/ResourceService.groovy:47]: personally prefer {} even there is just one line (currently)

File: grails-app/init/specimenbrowser/BootStrap.groovy (hunk 3 above)
Comment (charvolant) [reply grails-app/init/specimenbrowser/BootStrap.groovy:1]: This seems to be OK. There are a number of points in grails where it wants to use the app name as a package for configuration-type stuff and I figure it's best left alone. UrlMappings is another one.

File: grails-app/services/au/org/ala/specimenbrowser/ResourceService.groovy (hunk 4 above)
Comment (charvolant) [reply grails-app/services/au/org/ala/specimenbrowser/ResourceService.groovy:47]: Fixed.

**Pull request merged**